### PR TITLE
Restore CustomLauncher to Test Population

### PR DESCRIPTION
**Pull request by kyulee1**
Fixes https://github.com/dotnet/coreclr/issues/4076.
Custom launcher is needed for CoreRT tests.

### CODE REVIEW

**Comment by kyulee1** [comment]
@MichalStrehovsky PTAL

File: tests/src/CLRTest.Execute.Batch.targets [hunk 1]
```diff
@@ -229,9 +229,17 @@ IF NOT "%CLRTestExitCode%"=="%CLRTestExpectedExitCode%" (
 
     <PropertyGroup> 
       <_CLRTestRunFile Condition="'$(CLRTestIsHosted)'=='true'">"%CORE_ROOT%\corerun.exe"</_CLRTestRunFile>
+      <BatchCLRTestLaunchCmds><![CDATA[
+IF NOT "%CLRCustomTestLauncher%"=="" (
+  set LANUCHER=call %CLRCustomTestLauncher% %~dp0
```
Comment (MichalStrehovsky) [inline tests/src/CLRTest.Execute.Batch.targets]: Typo: LANUCHER

File: tests/src/CLRTest.Execute.Batch.targets [hunk 2]
```diff
@@ -229,9 +229,17 @@ IF NOT "%CLRTestExitCode%"=="%CLRTestExpectedExitCode%" (
...
+) ELSE (
+  set LANUCHER=%_DebuggerFullPath%
```
Comment (MichalStrehovsky) [inline tests/src/CLRTest.Execute.Batch.targets]: Copy paste typo: LANUCHER

File: tests/src/CLRTest.Execute.Batch.targets [hunk 3]
```diff
@@ -229,9 +229,17 @@ IF NOT "%CLRTestExitCode%"=="%CLRTestExpectedExitCode%" (
...
+)
+      ]]></BatchCLRTestLaunchCmds>
       <BatchCLRTestLaunchCmds Condition=" '$(IlasmRoundTrip)'=='true' "><![CDATA[
-ECHO $(_CLRTestRunFile) $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
-$(_CLRTestRunFile) $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
+$(BatchCLRTestLaunchCmds)
+ECHO %LAUNCHER% $(_CLRTestRunFile) $(TargetAssemblyName) %CLRTestExecutionArguments% %Host_Args%
```
Comment (MichalStrehovsky) [inline tests/src/CLRTest.Execute.Batch.targets]: $(_CLRTestRunFile) is set to corerun.exe, which means this generates something like

```
%LAUNCHER% "%CORE_ROOT%\corerun.exe" Finalizer.exe %CLRTestExecutionArguments% %Host_Args%
```

When it should generate
```
%LAUNCHER% Finalizer.exe %CLRTestExecutionArguments% %Host_Args%
```

Comment (MichalStrehovsky) [inline tests/src/CLRTest.Execute.Batch.targets]: Basically what the old code was going after was this line after all the expansion:

```
call %CLRCustomTestLauncher% %~dp0 Finalizer.exe %CLRTestExecutionArguments% %Host_Args%
```

First argument to the launcher is the path, second is the EXE name, and arguments follow.

**Comment by kyulee1** [comment]
Updated the change per feedback. PTAL.

**Comment by MichalStrehovsky** [comment]
:shipit:

Thank you! Having this back is super useful!

**Pull request merged**